### ISSUE TITLE
404 error loading module.js 

### ISSUE THREAD

**Issue by v0d0r** [kpfaulkner/sendgrid-plugin#2]
Tried installing this plugin for the first time.  I tried both releases both resulting in the same error.
My grafana version is 7.5.7.
I also tried renaming the folder in the plugins directory from just sendgrid-plugin/ to kpfaulkner-sendgrid-plugin/ with the same result. Any ideas?

'''Fetch error: 404 Not Found Instantiating http://10.0.0.1:3000/public/plugins/kpfaulkner-sendgrid-plugin/module.js Loading plugins/kpfaulkner-sendgrid-plugin/module'''

**Comment by kpfaulkner** [comment]
Hmmm, will try and repro later today. Can't say I've specifically tried 7.5.7 but would have expected it to still be fine. Will update here as soon as I can.

Thanks

Ken

**Comment by kpfaulkner** [comment]
Hi @v0d0r 

Didn't have any issues myself, so will need some extra details from you (and I freely admit my install instructions are a bit lacking)

1) Which OS are you using? 
2) Are you modifying the 2 entries in the defaults.ini?  One for the plugins directory ("plugins =") and one for "allow_loading_unsigned_plugins" ? 

Thanks

Ken

**Comment by v0d0r** [comment]
Hi @kpfaulkner 
Thanks for your quick response and effort.  After thinking about it during the night I had a brain fart and managed to fix it this morning. For what its worth and for anybody in future running into the same issue the details of my setup and issue was as follows.

OS: Centos 7.x
Grafana Version 7.5.7..  Installed via the yum repo.

Root Cause:  By default SELINUX is enabled on Centos and Redhat. Though this is a great security feature it can sometimes be a pain.  It was silently blocking the grafana user from serving the requested file resulting in a 404 error.  Usually this will not show up in any of the application logs. 

Fix/Workaround: A quick way around this (obviously not the recommended way as you compromise your security a bit) is to disable selinux.  1. check if its running with command "getenforce".   If enabled 2. Disable with command "setenforce 0"
A more safe and cleaner fix would be to create a selinux policy and/or fix selinux context etc.  I will look into this when I have more time.

Lastly on another note. This might be Sendgrid related rather than plugin related but do you know why the graph would only show data from about 12 hours ago?  Is this a timezone thing or a Sendgrid thing? I assume sendgrid stats is near real time and not just daily?

**Comment by kpfaulkner** [comment]
Hi @v0d0r 

Glad you got it sorted!

As for the graphs, the data available from the Sendgrid API is purely once a day. I'd *love* something realtime but haven't seen any APIs like that from them. The graphs SG have on their own portal are also every 24 hours... so maybe that's all that we have access to. For my needs (reason I wrote the plugin) this was enough, since I just needed to see weekly trends.

If I ever see them updating their API to something more fine grained, I'll modify accordingly.

Cheers

Ken
